### PR TITLE
Refactor vhost improvements 

### PR DESCRIPTION
**Pull request by trlinkin**

### CODE REVIEW

**Comment by hunner** [comment]
Does #26 satisfy this?

**Comment by hunner** [comment]
Offline comment was, why throw away all the work that `apache::vhost` can do? This method should actually work better for customers who want to modify the existing templates but not actually have to break the module.

**Comment by trlinkin** [comment]
How should we proceed on this one?

File: templates/vhost.conf.erb [hunk 1]
```diff
@@ -36,4 +36,9 @@
 <%= scope.function_template(['apache/vhost/_serveralias.erb']) -%>
 <%= scope.function_template(['apache/vhost/_setenv.erb']) -%>
 <%= scope.function_template(['apache/vhost/_ssl.erb']) -%>
-</VirtualHost>
+<% if @custom_fragment -%>
```
Comment (hunner) [inline templates/vhost.conf.erb:39]: Could you move this conditional logic into the `_custom_fragment_header.erb` file? Then the template fragment can call `scope.function_template(Array(@custom_fragment)` for all fragments that are passed as a potential array.

File: templates/vhost/_custom_fragment_header.erb [hunk 2]
```diff
@@ -0,0 +1,3 @@
+# ************************************
+#    Custom Fragment Section
+# ************************************
```
Comment (hunner) [inline templates/vhost/_custom_fragment_header.erb:3]: Could you make this comment consistent with the other vhost templates fragments?

``` conf
  ## Custom Fragments
```

File: templates/vhost.conf.erb [hunk 3]
```diff
@@ -36,4 +36,9 @@
...
+<% Array(@custom_fragment).each do |fragment| -%>
+<%= scope.function_template(['apache/vhost/_custom_fragment_header.erb',fragment]) %>
+<% end -%>
+<% end -%>
+<%= scope.function_template(['apache/vhost/_vhost_end.erb']) -%>
```
Comment (hunner) [inline templates/vhost.conf.erb:44]: Does this need to be broken out into its own template?

File: templates/vhost.conf.erb (hunk 1 above)
Comment (trlinkin) [inline templates/vhost.conf.erb:39]: Agreed

File: templates/vhost/_custom_fragment_header.erb (hunk 2 above)
Comment (trlinkin) [inline templates/vhost/_custom_fragment_header.erb:3]: I have no idea why I invented my own comment there. Consistency is better. +1

File: templates/vhost.conf.erb (hunk 3 above)
Comment (trlinkin) [inline templates/vhost.conf.erb:44]: No, I think I had a bigger idea that never came to be.

**Comment by trlinkin** [comment]
Closed since #36 Improves on this code and should be the PR getting merged.

**Pull request closed without merging**